### PR TITLE
Couple of fixes to make atmsort.pm happy

### PR DESCRIPTION
**Pull request by ashwinstar**
Fixes to correctly format the output for explain for few tests. Based on mailing list discussion https://groups.google.com/a/greenplum.org/d/msg/gpdb-dev/iPSs0a8S4X8/vM_tu_5KCAAJ.

### CODE REVIEW

File: src/test/regress/expected/subselect_gp.out [hunk 1]
```diff
@@ -997,9 +997,9 @@ select count(*) from
 (1 row)
 
        
----
---- Query was deadlocking because of not squelching subplans (MPP-18936)
----
+--
+-- Query was deadlocking because of not squelching subplans (MPP-18936)
+--
```
Comment (hsyuan) [inline src/test/regress/expected/subselect_gp.out]: also need to update subselect_gp_optimizer.out

**Comment by ashwinstar** [comment]
On Mon, Oct 2, 2017 at 5:31 PM, Haisheng Yuan <notifications@github.com>
wrote:

> *@hsyuan* commented on this pull request.
> ------------------------------
>
> In src/test/regress/expected/subselect_gp.out
> <https://github.com/greenplum-db/gpdb/pull/3436#discussion_r142289390>:
>
> > @@ -997,9 +997,9 @@ select count(*) from
>  (1 row)
>
>
> ----
> ---- Query was deadlocking because of not squelching subplans (MPP-18936)
> ----
> +--
> +-- Query was deadlocking because of not squelching subplans (MPP-18936)
> +--
>
> also need to update subselect_gp_optimizer.out
>
Thank You. Sure.

**Comment by dhanashreek89** [comment]
@ashwinstar I am seeing same issue in the PR pipeline for https://github.com/greenplum-db/gpdb/pull/5019. The fix in this PR touches the partition selector filter that is displayed in the `EXPLAIN` output.  The failure is observed for `EXPLAIN EXECUTE` and it shows differences in `Filter` and `Settings` which are normally ignored while diffing the EXPLAIN output.

I am planning on backporting this fix to 5X_STABLE as a part of the PR mentioned above, unless there were some reasons for not porting this. Thank you for fixing this!

**Comment by ashwinstar** [comment]
On Wed, May 23, 2018 at 9:30 AM, Dhanashree Kashid <notifications@github.com
> wrote:

> @ashwinstar <https://github.com/ashwinstar> I am seeing same issue in the
> PR pipeline for #5019 <https://github.com/greenplum-db/gpdb/pull/5019>.
> The fix in this PR touches the partition selector filter that is displayed
> in the EXPLAIN output. The failure is observed for EXPLAIN EXECUTE and it
> shows differences in Filter and Settings which are normally ignored while
> diffing the EXPLAIN output.
>
> I am planning on backporting this fix to 5X_STABLE as a part of the PR
> mentioned above, unless there were some reasons for not porting this. Thank
> you for fixing this!
>
We backport on need basis, so yes please feel free to do the same if the
need now arises.

**Pull request merged**